util/src/main/java/io/grpc/util/MultiChildLoadBalancer.java

```diff
@@ -188,6 +197,19 @@ private Status updateChildrenWithResolvedAddresses(
     return status;
   }
 
+  @VisibleForTesting
+  static <T> Iterable<T> offsetIterable(Collection<T> c, int seed) {
+    int pos;
+    if (c.isEmpty()) {
+      pos = 0;
+    } else {
+      pos = (seed & 0x7FFF_FFFF) % c.size();
```

Comment:
This is required such that `seed & 0x7FFF_FFFF` returns always a non-negative number (essentially removes the sign), since a negative number % some number returns a negative in Java and that is not ideal for array/iterable operations (e.g. `-1 % 10` returns `-1`).

Would it make more sense when generating the `OFFSET_SEED` we limit the bounds such that it is always non-negative?